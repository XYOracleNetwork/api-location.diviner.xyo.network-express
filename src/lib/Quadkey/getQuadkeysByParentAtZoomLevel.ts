import { getParentQuadkey } from './getParentQuadkey'
import { getQuadkeyAtZoomLevel } from './getQuadkeyAtZoomLevel'

export const getQuadkeysByParentAtZoomLevel = (heatmap: string[], zoom: number): Record<string, string[]> => {
  const quadkeysByParent = heatmap.reduce<Record<string, string[]>>((acc, curr) => {
    const quadkey = getQuadkeyAtZoomLevel(curr, zoom + 1)
    const parent = getParentQuadkey(quadkey)
    acc[parent] = acc[parent] || []
    acc[parent].push(quadkey)
    return acc
  }, {})
  return quadkeysByParent
}
