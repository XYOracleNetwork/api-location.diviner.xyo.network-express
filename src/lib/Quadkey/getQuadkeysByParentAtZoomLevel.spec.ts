import { getQuadkeysByParentAtZoomLevel } from './getQuadkeysByParentAtZoomLevel'

const testData = [
  { input: { quadkeys: ['00', '033', '0333'], zoom: 1 } },
  { input: { quadkeys: ['100', '1220', '12220'], zoom: 2 } },
  { input: { quadkeys: ['2000', '21100', '211100'], zoom: 3 } },
  { input: { quadkeys: ['30000', '300000', '3000000'], zoom: 4 } },
  { input: { quadkeys: ['400000', '400000', '400000'], zoom: 5 } },
]

describe('getQuadkeysByParent', () => {
  it.each(testData)('converts the quadkey at the specified zoom level', (data) => {
    const { quadkeys, zoom } = data.input
    const actual = getQuadkeysByParentAtZoomLevel(quadkeys, zoom)
    expect(actual).toMatchSnapshot()
  })
})
