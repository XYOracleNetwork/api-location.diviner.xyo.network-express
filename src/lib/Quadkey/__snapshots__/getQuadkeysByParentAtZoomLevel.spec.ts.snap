// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getQuadkeysByParent converts the quadkey at the specified zoom level 1`] = `
Object {
  "0": Array [
    "00",
    "03",
    "03",
  ],
}
`;

exports[`getQuadkeysByParent converts the quadkey at the specified zoom level 2`] = `
Object {
  "10": Array [
    "100",
  ],
  "12": Array [
    "122",
    "122",
  ],
}
`;

exports[`getQuadkeysByParent converts the quadkey at the specified zoom level 3`] = `
Object {
  "200": Array [
    "2000",
  ],
  "211": Array [
    "2110",
    "2111",
  ],
}
`;

exports[`getQuadkeysByParent converts the quadkey at the specified zoom level 4`] = `
Object {
  "3000": Array [
    "30000",
    "30000",
    "30000",
  ],
}
`;

exports[`getQuadkeysByParent converts the quadkey at the specified zoom level 5`] = `
Object {
  "40000": Array [
    "400000",
    "400000",
    "400000",
  ],
}
`;
